ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -438,46 +422,22 @@ public void testBadChannelPutFailure() throws Exception {
   }
 
   /**
-   * A bad arguments test, where the channel size is different from the size in BlobProperties.
-   */
-  @Test
-  public void testChannelSizeNotSizeInPropertiesPutFailure() throws Exception {
-    requestAndResultsList.clear();
-    int blobSize = chunkSize;
-    RequestAndResult requestAndResult = new RequestAndResult(blobSize);
-    // Change the actual content size.
-    requestAndResult.putContent = new byte[blobSize + 1];
-    requestAndResultsList.add(requestAndResult);
-    Exception expectedException = new RouterException("", RouterErrorCode.BadInputChannel);
-    submitPutsAndAssertFailure(expectedException, true, false);
-  }
-
-  /**
-   * A bad arguments test, where the blob size is very large.
+   * A bad arguments test, where the channel size is different from the size in BlobProperties. This should succeed
+   * as the size in the passed in BlobProperties is ignored by the router.
    */
   @Test
-  public void testBlobTooLargePutFailure() throws Exception {
```

Comment:
Yes, intentional. As you noticed in your second comment, unlike before when the error is caught at construction, now the router comes to know that a blob is too large only once it ends up writing those many chunks.